lib/ffi-libarchive-binary/libarchive_recipe.rb

```diff
@@ -36,11 +42,56 @@ def configure_defaults
       ]
     end
 
+    def configure
+      paths = [@zlib_recipe.path, @expat_recipe.path]
+      cflags = paths.map { |k| "-I#{k}/include" }.join(" ")
+      cmd = ["env", "CFLAGS=#{cflags}", "./configure"] + computed_options
+
+      execute("configure", cmd)
+
+      # drop default libexpat and zlib
+      libz = File.join(@zlib_recipe.path, "lib", "libz.a")
+      libexpat = File.join(@expat_recipe.path, "lib", "libexpat.a")
+
+      if LibarchiveBinary::windows?
+        # https://stackoverflow.com/a/14112368/902217
+        static_link_pref = "-Wl,-Bstatic,"
+        libz = libz.prepend(static_link_pref)
+        libexpat = libexpat.prepend(static_link_pref)
+      end
+
+      makefile = File.join(work_path, "Makefile")
+      replace_in_file(" -lz ", " #{libz} ", makefile)
+      replace_in_file(" -lexpat ", " #{libexpat} ", makefile)
+    end
+
+    def replace_in_file(search_str, replace_str, filename)
+      puts "Replace \"#{search_str}\" with \"#{replace_str}\" in #{filename}"
+
+      f = File.open(filename, "r")
+      content = f.read
+      f.close
+
+      content.gsub!(search_str, replace_str)
+
+      File.open(filename, "w") { |f| f << content }
+    end
+
+    def activate
+      @zlib_recipe.activate
+      @expat_recipe.activate
+
+      super
+    end
+
     def cook_if_not
       cook unless File.exist?(checkpoint)
     end
 
     def cook
+      @zlib_recipe.cook_if_not
+      @expat_recipe.cook_if_not
+
       super
 
       FileUtils.touch(checkpoint)
```

Comment:
There is an `install` method which copies the libs to the place where they would be packaged to a gem. Since we use a new extension `.a`, we need to change this place.

```ruby
    def install
      super
      libs = Dir.glob(File.join(port_path, "{lib,bin}", "*"))
        .grep(/\/(?:lib)?[a-zA-Z0-9\-]+\.(?:so|dylib|dll)$/)
      FileUtils.cp_r(libs, ROOT.join("lib", "ffi-libarchive-binary"), verbose: true)
    end
```

And also new libs should be added to a runtime-generated gemspecs in `Rakefile` (`spec.files`):

```ruby
  task "platform:#{platform}" do
    spec = Gem::Specification::load("ffi-libarchive-binary.gemspec").dup
    spec.platform = Gem::Platform.new(platform)
    spec.files += ["lib/ffi-libarchive-binary/#{lib}"]
```

Reply:
I think we don't need this because we build `zlib` and `libexpat` as a static library and link them statically, this means that both `zlib` and `libexpat` become embed into the final libarchive.{so|dylib|dll}

Nice summary about both static vs runtime linking https://stackoverflow.com/a/61140381/902217

Reply:
Oh, thank you, I'll dive into it :)